xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp

```diff
@@ -463,6 +460,11 @@ void CDVDSubtitlesLibass::ApplyStyle(const std::shared_ptr<struct style>& subSty
       style->Shadow = 4 * scale; // Space between the text and the box edges
     }
 
+    double lineSpacing;
+    if (subStyle->assOverrideFont)
+      lineSpacing = subStyle->lineSpacing * scaleDefault;
+    else
+      lineSpacing = subStyle->lineSpacing * scale;
```

Comment:
here there is an error
you are calculating/set the linespace with a percentage value instead of px and so its wrong,
imo a good solution is to proportion the percentage value to half the height of the frame
so that when is to 100% the space reaches at max the half screen

here the change:
```suggestion
    double lineSpacing = (static_cast<double>(subStyle->lineSpacing) * (playResY / 2)) / 100;
    if (subStyle->assOverrideFont)
      lineSpacing *= scaleDefault;
    else
      lineSpacing *= scale;

    // ass_set_line_spacing do not scale, so we have to scale to frame size
    ass_set_line_spacing(m_renderer,
                         lineSpacing / playResY * static_cast<double>(opts.frameHeight / 2));
```

Reply:
after this change check again if the default value is correct for the box problem

Reply:
Good idea, although I think that limiting to a quarter of the height is better, otherwise the steps are too big. I've applied the change and adjusted the default.

Reply:
it seems a little limiting to use `/ 4` but i don't have much preference for that
from my pov i don't see the point of applying too much line space
is probably enough